core/domain/feature_flag_registry.py

```diff
@@ -0,0 +1,291 @@
+                feature_flag.name,
+                feature_flag.feature_flag_config.force_enable_for_all_users,
+                feature_flag.feature_flag_config.rollout_percentage,
+                feature_flag.feature_flag_config.user_group_ids
+            )
+            return
+
+        model_instance.force_enable_for_all_users = (
+            feature_flag.feature_flag_config.force_enable_for_all_users)
+        model_instance.rollout_percentage = (
+            feature_flag.feature_flag_config.rollout_percentage)
+        model_instance.user_group_ids = (
+            feature_flag.feature_flag_config.user_group_ids)
+        model_instance.update_timestamps()
+        model_instance.put()
+
+
+feature_flag_name_enum_to_description = {
+    FeatureNames.DUMMY_FEATURE_FLAG_FOR_E2E_TESTS: (
+        'This is a dummy feature flag for the e2e tests.'
+    ),
+    FeatureNames.END_CHAPTER_CELEBRATION: (
+        'This flag is for the end chapter celebration feature.'
+    ),
+    FeatureNames.CHECKPOINT_CELEBRATION: (
+        'This flag is for the checkpoint celebration feature.'
+    ),
+    FeatureNames.CONTRIBUTOR_DASHBOARD_ACCOMPLISHMENTS: (
+        'This flag enables showing per-contributor accomplishments on the '
+        'contributor dashboard.'
+    ),
+    FeatureNames.ANDROID_BETA_LANDING_PAGE: (
+        'This flag is for Android beta promo landing page.'
+    ),
+    FeatureNames.BLOG_PAGES: (
+        'This flag is for blog home page, blog author profile page and blog '
+        'post page.'
+    ),
+    FeatureNames.DIAGNOSTIC_TEST: (
+        'This flag is for the diagnostic test functionality.'
+    ),
+    FeatureNames.SERIAL_CHAPTER_LAUNCH_CURRICULUM_ADMIN_VIEW: (
+        'This flag is for serial chapter launch feature and making changes '
+        'only in the curriculum admin view.'
+    ),
+    FeatureNames.SERIAL_CHAPTER_LAUNCH_LEARNER_VIEW: (
+        'This flag is for serial chapter launch feature and making changes '
+        'only in the learner view.'
+    ),
+    FeatureNames.SHOW_REDESIGNED_LEARNER_DASHBOARD: (
+        'This flag is to show redesigned learner dashboard.'
+    ),
+    FeatureNames.SHOW_TRANSLATION_SIZE: (
+        'This flag is to show translation size on translation cards in '
+        'contributor dashboard.'
+    ),
+    FeatureNames.SHOW_FEEDBACK_UPDATES_IN_PROFILE_PIC_DROPDOWN: (
+        'This flag is to show feedback updates in the '
+        'profile pic drop-down menu.'
+    ),
+    FeatureNames.CD_ADMIN_DASHBOARD_NEW_UI: (
+        'This flag is to show new contributor admin dashboard.'
+    ),
+    FeatureNames.IS_IMPROVEMENTS_TAB_ENABLED: (
+        'Exposes the Improvements Tab for creators in the exploration editor.'
+    ),
+    FeatureNames.LEARNER_GROUPS_ARE_ENABLED: 'Enable learner groups feature'
+}
+
+for feature_flag_name_enum_in_dev in platform_feature_list.DEV_FEATURES_LIST:
+    Registry.create_feature_flag(
```

Comment:
Is this run everytime a request hits the server? Can it get run more than once? (Check using logging.)

Also I assume that, for this registry to be useful, we want to get the actual feature flag configs and not placeholders. Shouldn't that require datastore calls? Ideally it should be a get_multi call to fetch all the configs in one network call.

Reply:
Yes, this gets run more than once. This is fine because what it mainly does is updates the `featuer_flag_spec_registry` variable with the static values.

I have explained above why should we not make the calls to the backend [here](https://github.com/oppia/oppia/pull/18819#discussion_r1434027976).